### PR TITLE
Replace container logs(docker REST endpoint) with custom tail plugin 

### PR DESCRIPTION
**Pull request by bragi92**
Goes hand in hand with OMI code where we need to mount the container log volume and add the NOPASSWD requirement for "customtailfilereader.rb"

### CODE REVIEW

File: installer/conf/container.conf [hunk 1]
```diff
@@ -52,12 +52,11 @@
 
 # Container log
 <source>
-	type omi
-	run_interval 300s
+	type custom_sudo_tail
```
Comment (samisms) [inline installer/conf/container.conf]: Rename to containerlog_sudo_tail

File: installer/datafiles/base_container.data [hunk 2]
```diff
@@ -23,10 +23,13 @@ MAINTAINER:              'Microsoft Corporation'
 
 /opt/microsoft/omsagent/plugin/filter_docker_log.rb;			source/code/plugin/filter_docker_log.rb;			644; root; root
 /opt/microsoft/omsagent/plugin/filter_container.rb;			    source/code/plugin/filter_container.rb;				644; root; root
+/opt/microsoft/omsagent/plugin/filter_container_log.rb;			source/code/plugin/filter_container_log.rb;		    644; root; root
 
 /opt/microsoft/omsagent/plugin/in_kube_podinventory.rb;			source/code/plugin/in_kube_podinventory.rb;			644; root; root
 /opt/microsoft/omsagent/plugin/in_kube_events.rb;			    source/code/plugin/in_kube_events.rb;				644; root; root
 /opt/microsoft/omsagent/plugin/KubernetesApiClient.rb;			source/code/plugin/KubernetesApiClient.rb;			644; root; root
+/opt/microsoft/omsagent/plugin/in_custom_sudo_tail.rb;			source/code/plugin/in_custom_sudo_tail.rb;			644; root; root
```
Comment (samisms) [inline installer/datafiles/base_container.data]: change once renamed

File: source/code/plugin/customtailfilereader.rb [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+      end
+
+      attr_reader :path
+
+      def wrap_receive_lines(lines)
+	newLines = []
+	containerID = @path.split('/').last.chomp('-json.log')
+	containerInspectInformation = "/var/opt/microsoft/docker-cimprov/state/ContainerInventory/" + containerID
+	tempContainerInfo = {}
+	File.open(containerInspectInformation) { |f| tempContainerInfo = JSON.parse(f.readline)}
```
Comment (samisms) [inline source/code/plugin/customtailfilereader.rb]: Add exception handling , for file open fail, json parse fail and catch all other exceptions

File: source/code/plugin/filter_container_log.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+module Fluent
+  require 'logger'
+  class PassThruFilter < Filter
+    Fluent::Plugin.register_filter('filter_container_log', self)
+
+    def configure(conf)
+      super
+      require_relative 'omslog'
+      @containerCache = Hash.new
```
Comment (samisms) [inline source/code/plugin/filter_container_log.rb]: Remove if not using

File: source/code/plugin/in_custom_sudo_tail.rb [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        OMS::Log.warn_once("#{path} is not readable. Cannot tail the file.")
+	return ""
+      end
+    end
+
+    def set_system_command
+      date = Time.now
+      cName = "Unkown"
+      tempContainerInfo = {}
+      path = "/var/opt/microsoft/docker-cimprov/state/ContainerInventory/*"
```
Comment (samisms) [inline source/code/plugin/in_custom_sudo_tail.rb]: Add a class variable for the path

**Comment by bragi92** [comment]
@samisms Updated PR with requested changes.

File: source/code/plugin/containerlogtailfilereader.rb [hunk 6]
```diff
@@ -76,14 +76,15 @@ def initialize(path, pe, read_from_head, log, &receive_lines)
...
+        @containerIDFilePath = "/var/opt/microsoft/docker-cimprov/state/ContainerInventory/"
       end
 
       attr_reader :path
 
       def wrap_receive_lines(lines)
 	newLines = []
 	containerID = @path.split('/').last.chomp('-json.log')
-	containerInspectInformation = "/var/opt/microsoft/docker-cimprov/state/ContainerInventory/" + containerID
+	containerInspectInformation = @containerIDFilePath + containerID
```
Comment (samisms) [inline source/code/plugin/containerlogtailfilereader.rb]: Can you add comments describing the docker log file location . Just to make it more readable and easy to maintain ?

File: source/code/plugin/customtailfilereader.rb (hunk 3 above)
Comment (samisms) [reply source/code/plugin/customtailfilereader.rb]: Add exception handling for this.

File: installer/conf/container.conf [hunk 7]
```diff
@@ -52,7 +52,7 @@
 
 # Container log
 <source>
-	type custom_sudo_tail
+	type containerlog_sudo_tail
```
Comment (samisms) [inline installer/conf/container.conf]: Add a sample line from the file so your "format /\"log\"=>\"(?<LogEntry>.*)", \"stream\"=>\"(?<LogEntrySource>.*)", \"time\"=>\"(?<TimeGenerated>.*)", \"ContainerID\"=>\"(?<Id>.*)", \"Image\"=>\"(?<Image>.*)", \"Name\"=>\"(?<Name>.*)", \"SourceSystem\"=>\"(?<SourceSystem>.*)"}/" is easy to understand.

**Pull request merged**